Add UI for movie details Feature

HendSayed25 opened the pull request:
# What's New ?

1- Add Design For Movie Details Screen and Review Screen
2- Use View Model in Movie Details Screen
3- Fix some issues in details components

HendSayed25 committed:
Merge remote-tracking branch 'origin/develop' into feature/movie-details

HendSayed25 committed:
fix preview issue

HendSayed25 committed:
edit value of padding

HendSayed25 committed:
add initial design in details screen

HendSayed25 committed:
Merge remote-tracking branch 'origin/feature/movie-details' into feature/movie-details

# Conflicts:
#	presentation/details/build.gradle.kts
#	presentation/details/src/main/res/values-ar-rEG/strings.xml
#	presentation/details/src/main/res/values/strings.xml

HendSayed25 committed:
fix problem in padding value

HendSayed25 committed:
edit some design in movie details screen

HendSayed25 committed:
Merge remote-tracking branch 'origin/feature/movie-details' into feature/movie-details

# Conflicts:
#	presentation/details/src/main/res/values-ar-rEG/strings.xml
#	presentation/details/src/main/res/values/strings.xml

HendSayed25 committed:
convert any hard coded in string file

HendSayed25 committed:
Merge remote-tracking branch 'origin/feature/movie-details' into feature/movie-details

# Conflicts:
#	presentation/details/src/main/res/values-ar-rEG/strings.xml

HendSayed25 committed:
add image for test

HendSayed25 committed:
add design for movieReview

HendSayed25 committed:
edit error in string file

HendSayed25 committed:
add modifiers for icon

HendSayed25 committed:
convert from LocalDate To LocalDateTime recording to entity

HendSayed25 committed:
convert from LocalDate To LocalDateTime recording to entity

HendSayed25 committed:
edit name of string resource

HendSayed25 committed:
edit design

HendSayed25 committed:
convert from local date to local date time

HendSayed25 committed:
add details view model in di

HendSayed25 committed:
edit view model and states

HendSayed25 committed:
remove unused section

HendSayed25 committed:
add needed clicks and handle image

HendSayed25 committed:
add needed component and image

HendSayed25 committed:
edit background color for bottom sheet

HendSayed25 committed:
connect view model with movie details screen

HendSayed25 committed:
fix issue in preview

HendSayed25 committed:
Merge remote-tracking branch 'origin/develop' into feature/movie-details-ui

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/viewModelModule.kt
#	presentation/details/src/main/java/com/giraffe/details/components/MainMovieOrSeriesDetails.kt
#	presentation/details/src/main/java/com/giraffe/details/components/ReviewCard.kt
#	presentation/details/src/main/java/com/giraffe/details/components/StarCastSection.kt
#	presentation/details/src/main/java/com/giraffe/details/preview/PreviewDetails.kt
#	presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/MovieDetailsScreenState.kt

HendSayed25 committed:
update screen

HendSayed25 committed:
update screen state

HendSayed25 committed:
fix errors

HendSayed25 committed:
fix problem of rate button

HendSayed25 committed:
change location of screen

HendSayed25 committed:
Merge remote-tracking branch 'origin/develop' into feature/movie-details-ui

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/viewModelModule.kt

HendSayed25 committed:
add new texts to string file

HendSayed25 committed:
add details view model in di

HendSayed25 committed:
add content padding to button

HendSayed25 committed:
update the type of list of crew

HendSayed25 committed:
put the clicking in whole card not only the icon

HendSayed25 committed:
update some structure in staff section

HendSayed25 committed:
updates the preview functions for the `StaffInfoSection`

HendSayed25 committed:
Refactor: Improve MovieDetails screen UI and padding

mohannadahmed00 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/AppBar.kt:
```diff
@@ -56,13 +56,20 @@ fun AppBar(
             image = image,
-            imageSize = imageSize
+            imageSize = imageSize,
+            modifier = Modifier.padding(start = 8.dp)
```
is this padding in design system? or you just need it

HendSayed25 replied:
yes, in design system and missed in our component

mohannadahmed00 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/button_type/PrimaryButton.kt:
```diff
@@ -37,6 +39,7 @@ fun PrimaryButton(
             containerColor = buttonColor.value,
             contentColor = if (enabled) Theme.color.button.onPrimary else Theme.color.button.onDisabled,
         ),
+        contentPadding = PaddingValues(vertical = 16.dp)
```
is this padding in design system? or you just need it

HendSayed25 replied:
yes , all buttons in screens has this padding

Mohamed-Ashraf-2210 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/AppBar.kt:
```diff
@@ -56,13 +56,20 @@ fun AppBar(
             image = image,
-            imageSize = imageSize
+            imageSize = imageSize,
+            modifier = Modifier.padding(start = 8.dp)
```
already exist spaced by arrangement this padding not necessary

Mohamed-Ashraf-2210 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/AppBar.kt:
```diff
@@ -56,13 +56,20 @@ fun AppBar(
-            Title(title)
+            Caption(
+                caption=caption,
+                modifier= Modifier.padding(start = 8.dp)
```
already exist spaced by arrangement this padding not necessary

Mohamed-Ashraf-2210 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/AppBar.kt:
```diff
@@ -56,13 +56,20 @@ fun AppBar(
+            )
+            Title(
+                title = title,
+                modifier= Modifier.padding(start = 8.dp)
```
already exist spaced by arrangement this padding not necessary

HendSayed25 replied:
the space without it so small not like design

Mohamed-Ashraf-2210 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/BaseBottomSheet.kt:
```diff
@@ -43,7 +43,8 @@ fun BaseBottomSheet(
             shape = RoundedCornerShape(Theme.radius.xl),
-            contentColor = Theme.color.background.bottomSheet
+            contentColor = Theme.color.background.bottomSheet,
+            containerColor = Theme.color.background.bottomSheet
```
unused variable pls remove this containerColor

HendSayed25 replied:
when remove it the color of bottom sheet still white when using dark theme , 
so when i put it it's color become changed recording to theme

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/components/CollectionItem.kt:
```diff
@@ -0,0 +1,88 @@
+
+@Composable
+fun CollectionItem(
+    modifier: Modifier = Modifier,
```
modifier is a first parameter not requested

HendSayed25 replied:
what you mean?

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/components/RatingSection.kt:
```diff
@@ -27,7 +27,7 @@ import com.giraffe.designsystem.theme.Theme
     modifier: Modifier = Modifier,
     rate: Int? = null,
-    onArrowButtonClick: () -> Unit = {}
+    onCardClick: () -> Unit = {}
```
can you naming onClick only

HendSayed25 replied:
i use this name because this component i can put the click in different components in it so i specifiy it

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/components/StaffInfoSection.kt:
```diff
@@ -43,30 +39,21 @@ fun StaffInfoSection(
+                    StaffItem(name = name, role = key)
 
-                    if (index != staffList.lastIndex) {
+                    if (index != staffList.entries.size - 1) {
```
replace if to AnimatedVisibility

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieDetails.kt:
```diff
@@ -0,0 +1,239 @@
+
+
+@Composable
+fun MovieDetails(
```
MovieDetailsScreen*

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieDetails.kt:
```diff
@@ -0,0 +1,239 @@
+    onDismissAddRatingBottomSheet: () -> Unit,
+) {
+
+    LazyColumn(
```
you can use column instead of lazyColumn

HendSayed25 replied:
what is the problem of using lazy?

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieDetails.kt:
```diff
@@ -0,0 +1,239 @@
+        }
+
+        itemsIndexed(state.movieReviews.take(3)) { index, review ->
+            val padding = when (index) {
```
use animateDpAsState to make your screen smooth

HendSayed25 replied:
using animated when has one component and his shape will change recording to specific change right?
here i take the instance from it not use the same instance

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieReviewsScreen.kt:
```diff
@@ -0,0 +1,154 @@
+import com.giraffe.media.movies.entity.MovieReviewAuthor
+
+@Composable
+fun MovieReviewsScreen(
```
rename to ReviewsScreen only to use in series review

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieReviewsScreen.kt:
```diff
@@ -0,0 +1,154 @@
+
+@Composable
+fun MovieReviewsScreen(
+    movieReviewList : List<MovieReview>
```
rename it to reviews and make this list fit to series and movie review together

HendSayed25 replied:
how determine the type of list?

Mohamed-Ashraf-2210 replied:
make List<Review> it is the same in series and movie, it is a review why you sepeted it

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieReviewsScreen.kt:
```diff
@@ -0,0 +1,154 @@
+    movieReviewList : List<MovieReview>
+){
+
+    MovieReviewsContent(
```
make it to fit to series and movie together

HendSayed25 replied:
only change the name i understand 😂

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieDetails.kt:
```diff
@@ -0,0 +1,239 @@
+        }
+        item {
+            MainMovieOrSeriesDetails(
+                type = "Movie",
```
remove hard coded

Mohamed-Ashraf-2210 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/screens/MovieDetails.kt:
```diff
@@ -0,0 +1,239 @@
+            ReviewCard(
+                modifier = padding,
+                rate = review.author.rating.toInt(),
+                reviewText = review.content,
```
rename reviewText to contentReview

HendSayed25 committed:
Change PrimaryButton vertical padding

HendSayed25 committed:
Rename MovieDetails composable to MovieDetailsScreen

HendSayed25 committed:
Rename MovieDetails to MovieDetailsScreen

HendSayed25 committed:
Rename MovieReviewsScreen to ReviewsScreen

HendSayed25 committed:
rename `onCardClick` to `onClickCard`

HendSayed25 committed:
Adjust padding in AppBar composable

HendSayed25 committed:
Reorder parameters in CollectionItem composable

HendSayed25 committed:
Add "Movie" string resource

HendSayed25 committed:
Refactor: Use string resources and add padding to bottom sheets

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/details/components/CollectionItem.kt:
it already exists in design system

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/MovieDetailsViewModel.kt:
```diff
@@ -85,11 +95,23 @@ class MovieDetailsViewModel(
         val cast = people.filter { it.role == "Acting" }
         val crew = people.filter { it.role != "Acting" }
+        val mappedCrew = crew.map { person ->
+            StaffMember(
```
use ui mapper here to map from person into StaffMember

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/details/screens/moviedetails/MovieDetailsViewModel.kt:
```diff
@@ -85,11 +95,23 @@ class MovieDetailsViewModel(
-                cast = cast,
-                crew = crew,
+                cast = cast.map { person->
+                    CastMember(
```
use ui mapper here to map from person into CastMember

HendSayed25 committed:
feat: remove unused collection item and domain entities

HendSayed25 committed:
feat: Add review and crew UI ,cast models

HendSayed25 committed:
Refactor: Convert MovieUi to/from Entity mapping to extension functions

HendSayed25 committed:
Refactor: Use generic Review entity

HendSayed25 committed:
Merge remote-tracking branch 'origin/develop' into feature/movie-details-ui

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/preview/PreviewDetails.kt
#	presentation/details/src/main/res/values-ar-rEG/strings.xml
#	presentation/details/src/main/res/values/strings.xml
#	repository/media/src/main/java/com/giraffe/media/movie/MoviesRepositoryImpl.kt

HendSayed25 committed:
Merge remote-tracking branch 'origin/develop' into feature/movie-details-ui

HendSayed25 committed:
Refactor: Adjust padding in AddToCollectionContent and update StarCastSection

HendSayed25 committed:
Refactor series reviews mapper
The `parseData` function has been updated to return `LocalDateTime` instead of `LocalDate`.

HendSayed25 committed:
Refactor: Update Review entity path in use cases

HendSayed25 committed:
Refactor: Use UI models in MovieDetails ViewModel

HendSayed25 committed:
Refactor MovieReviewMapper to use a generic Review entity

HendSayed25 committed:
Refactor: Simplify getMovieDetails method in MoviesRepositoryImpl

HendSayed25 committed:
Refactor ReviewsScreen to use ReviewUI model

HendSayed25 committed:
feat: display movie reviews in movie details screen

HendSayed25 committed:
Migrate SeriesReview to Review in fakeSeries

HendSayed25 committed:
Refactor: Update GetMovieReviewsUseCaseTest to use common Review entity
